src/network/core/tcp_game.h

```diff
@@ -203,6 +203,7 @@ class NetworkGameSocketHandler : public NetworkTCPSocketHandler {
 	 * Send information about a client:
 	 * uint32_t  ID of the client (always unique on a server. 1 = server, 0 is invalid).
 	 * uint8_t   ID of the company the client is playing as (255 for spectators).
+	 * uint16_t  Bitmask of the companies this client can join.
```

Comment:
As the allow-list is stored in the savegame later on, isn't this information deducible from the savegame?

I am mainly considering the case where some patch-sets make it possible to have 500 companies in network games. This would be another nail in that idea, as having 500 bits is impractical :)

Reply:
Btw, for some ideas I wrote in https://gist.github.com/TrueBrain/cd5979cb7ba14b6ac649c74cd9fcd3eb, we would need to get the pubkey of all clients to all clients. So that might be good to use here, as that would also allow the DoCommand to add any other pubkey they like (which for example they got via a copy/paste from a friend).

I know, not for this PR, but just some food for thought.

Reply:
The pool has a decent bitmap with way more than 500 bits.
Also the stored public key strings are at least 65 * 8 bits (520).

But alas, I do not think we should send the public key with the client to all clients as it's essentially a unique identifier that you might want to share with the server but not with everybody.
It's not that I'd like to share my SSH public key really publicly, I share it only with servers I want to login to.

Reply:
Hmm, that is not how public keys work :) They are meant to be shared, to be public. For example:

https://github.com/Rubidium42.keys

You already share publicly your SSH public key (sorry if you didn't know :D ).

But mostly, you already send the public keys of everyone on the allow list to all clients. So that is already done / implied :)